src/lib/storage/table.cpp

```diff
@@ -236,6 +238,16 @@ void Table::append_chunk(const Segments& segments, std::shared_ptr<MvccData> mvc
       const auto is_reference_segment = std::dynamic_pointer_cast<ReferenceSegment>(segment) != nullptr;
       Assert(is_reference_segment == (_type == TableType::References), "Invalid Segment type");
     }
+
+    // Check that existing chunks are not empty
+    const auto chunk_count = _chunks.size();
+    for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
```

Comment:
Wouldn't it be sufficient to check if only the last existing chunk is empty?

Reply:
Probably, but since this is done only for debug builds, we don't lose anything.